src/main/java/be/seeseemelk/mockbukkit/inventory/meta/BannerMetaMock.java

```diff
@@ -0,0 +1,86 @@
+package be.seeseemelk.mockbukkit.inventory.meta;
+
+import org.bukkit.DyeColor;
+import org.bukkit.block.banner.Pattern;
+import org.bukkit.inventory.meta.BannerMeta;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BannerMetaMock extends ItemMetaMock implements BannerMeta
+{
+    private DyeColor baseColor; // defaults to null
+    private List<Pattern> patterns = new ArrayList<>();
+
+    public BannerMetaMock()
+    {
+        super();
+    }
+
+    public BannerMetaMock(ItemMeta meta)
+    {
+        super(meta);
+
+        if (meta instanceof BannerMeta) {
+            BannerMeta bannerMeta = (BannerMeta) meta;
+            this.setBaseColor(((BannerMeta) meta).getBaseColor());
+            this.setPatterns(((BannerMeta) meta).getPatterns());
+        }
+
+    }
+
+    @Override
+    public DyeColor getBaseColor()
+    {
+        return this.baseColor;
+    }
+
+    @Override
+    public void setBaseColor(DyeColor color)
+    {
+        this.baseColor = color;
+    }
+
+    @Override
+    public List<Pattern> getPatterns()
+    {
+        return new ArrayList<>(patterns);
```

Comment:
Why are you returning a new `ArrayList` here? It might be useful to add a comment noting that this is inline with CraftBukkit because it looks a bit weird.